### PR TITLE
WIP: Use boost timer with 100 connection limit

### PR DESCRIPTION
**Pull request by gtmills**
WIP: Need to test.

This PR
Reverts https://github.com/ibm-openbmc/bmcweb/commit/6dc5b8923fcecb3d3062e2b9a127262e4bfd26ed
Cherry picks the following 3 merged commits from upstream:
https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/49318
https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/49380/
https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/49380/

I find this solution best because
    These have merged upstream and is the direction upstream is headed.  
    It simplifies the logic over the current timer queue 
    I find bumping the current timer queue from 100 to 300 a workaround, a bit of a hack
    We should have a connection limit

### CODE REVIEW

File: http/http_connection.hpp [hunk 1]
```diff
@@ -300,10 +293,6 @@ class Connection :
                                         const boost::system::error_code& ec) {
                                         if (ec)
                                         {
-                                            BMCWEB_LOG_ERROR
```
Comment (geissonator) [inline http/http_connection.hpp:303]: Hmm, was this a useful error that we should keep?

File: http/timer_queue.hpp [hunk 2]
```diff
@@ -43,7 +43,6 @@ class TimerQueue
     {
         if (dq.size() == maxSize)
         {
-            BMCWEB_LOG_ERROR << "timer add: dq.size has hit maxSize";
```
Comment (geissonator) [inline http/timer_queue.hpp:46]: Still want this message?

File: http/http_connection.hpp [hunk 3]
```diff
@@ -278,6 +274,12 @@ class Connection :
 
     void start()
     {
+        if (connectionCount >= 100)
+        {
+            BMCWEB_LOG_CRITICAL << this << "Max connection count exceeded.";
```
Comment (geissonator) [inline http/http_connection.hpp:279]: Ahh, we get the count limit message here so we're good on that one.

File: http/timer_queue.hpp (hunk 2 above)
Comment (gtmills) [reply http/timer_queue.hpp:46]: if (connectionCount >= 100)
        {
            BMCWEB_LOG_CRITICAL << this << "Max connection count exceeded.";
            
            
should cover us. 
This use case isn't really valid anymore.

File: http/http_connection.hpp (hunk 3 above)
Comment (gtmills) [reply http/http_connection.hpp:279]: Yeah.

File: http/http_connection.hpp (hunk 1 above)
Comment (gtmills) [reply http/http_connection.hpp]: It didn't hurt so added back. Done.

**Review by geissonator — APPROVED** [review]

**Pull request merged**